overlord/snapstate/snapmgr.go

```diff
@@ -526,6 +527,61 @@ func (m *SnapManager) ensureRefreshes() error {
 	return nil
 }
 
+// ensureForceDevmodeDropsDevmodeFromState undoes the froced devmode
+// in snapstate for forced devmode distros.
+func (m *SnapManager) ensureForceDevmodeDropsDevmodeFromState() error {
+	if !release.ReleaseInfo.ForceDevMode() {
+		return nil
+	}
+
+	m.state.Lock()
+	defer m.state.Unlock()
+
+	// int because we might want to come back and do a second pass at cleanup
+	var fixed int
+	if err := m.state.Get("fix-forced-devmode", &fixed); err != nil && err != state.ErrNoState {
+		return err
+	}
+
+	if fixed > 0 {
+		return nil
+	}
+
+	var snaps map[string]*json.RawMessage
+	if err := m.state.Get("snaps", &snaps); err != nil {
+		if err == state.ErrNoState {
```

Comment:
This is nice, but I'm not sure we really need it. If we have no snaps the loop below will not have any effect. So we could test ` ; err != nil && err != state.ErrNoState {`. But not super important (i.e. nitpick land).